statsmodels/base/optimizer.py

```diff
@@ -217,6 +217,36 @@ def _fit_regularized(self, params):
 ########################################
 # Helper functions to fit
 
+
+def _fit_minimize(f, score, start_params, fargs, kwargs, disp=True,
+                        maxiter=100, callback=None, retall=False,
+                        full_output=True, hess=None):
+    kwargs.setdefault('min_method', None)
+
+    # prepare options dict for minimize
+    filter_opts = ['extra_fit_funcs', 'niter', 'min_method', 'tol']
+    options = {k:v for k,v in kwargs.items() if k not in filter_opts}
+    options['disp']    = disp
+    options['maxiter'] = maxiter
+
+    # we're using BFGS by default, which doesn't need hessian
+    if kwargs['min_method'] == None: hess = None
+
+    res = optimize.minimize(f, start_params, args=fargs, method=kwargs['min_method'],
+                            jac=score, hess=hess, callback=callback, options=options)
+
+    xopt    = res.x
+    retvals = None
+    if full_output:
+        retvals = {'fopt': res.fun, 'iterations': res.nit,
```

Comment:
since it's just one line, we can add scipy 0.14 compatibility, AFAICS
use np.nan if res.nit is not available
`nit = getattr(res, 'nit', np.nan)`